extensions/notebook/src/extension.ts

```diff
@@ -25,6 +26,11 @@ let controller: JupyterController;
 type ChooseCellType = { label: string, id: CellType };
 
 export async function activate(extensionContext: vscode.ExtensionContext): Promise<IExtensionApi> {
+
+	const bookTreeViewProvider = new BookTreeViewProvider(vscode.workspace.rootPath || '');
```

Comment:
>|| '' [](start = 81, length = 5)

Does this handle switching workspaces correctly? (Open Book, switch workspaces - does it still function correctly? And can you open up a different book from a different location?)

Reply:
Switching workspaces causes a full restart of the app. Each workspace -> 1 Azure Data Studio window. On changing workspace, you get a restart of the extension host